daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -66,54 +73,165 @@ final case class ScenarioRunner(
       res match {
         case SResultFinalValue(v) =>
           finalValue = v
+
         case SResultError(err) =>
           throw SRunnerException(err)
 
-        case SResultNeedPackage(pkgId, _) =>
-          crash(s"package $pkgId not found")
-
-        case SResultNeedContract(coid, tid @ _, committers, cbMissing, cbPresent) =>
-          // We never have readAs parties in scenarios.
-          lookupContractUnsafe(coid, committers, Set.empty, cbMissing, cbPresent)
-
         case SResultNeedTime(callback) =>
           callback(ledger.currentTime)
 
-        case SResultScenarioMustFail(tx, committers, callback) =>
-          // We never have readAs parties in scenarios.
-          mustFail(tx, committers, Set.empty)
-          callback(())
-
-        case SResultScenarioCommit(value, tx, committers, callback) =>
-          // We never have readAs parties in scenarios.
-          commit(value, tx, committers, Set.empty, callback)
-
         case SResultScenarioPassTime(delta, callback) =>
           passTime(delta, callback)
 
-        case SResultScenarioInsertMustFail(committers, optLocation) => {
-          // We never have readAs parties in scenarios.
-          ledger = ledger.insertAssertMustFail(committers, Set.empty, optLocation)
-        }
-
         case SResultScenarioGetParty(partyText, callback) =>
           getParty(partyText, callback)
 
-        case SResultNeedKey(keyWithMaintainers, committers, cb) =>
-          // We never have readAs parties in scenarios.
-          lookupKeyUnsafe(keyWithMaintainers.globalKey, committers, Set.empty, cb)
+        case SResultScenarioSubmit(committers, commands, location, mustFail, callback) =>
+          val submitResult = submit(committers, commands, location)
+          // TODO (MK) We copy the ptx & commit location and the trace
```

Comment:
This is definitely not the desired endstate. I’d like to keep it for a separate PR though. Fixing this really only makes sense if we make the same change in Daml script (which currently does something similarly messy because it had to fit in what the scenario service did :facepalm:) and this PR is large enough.